extensions/amp-delight-player/0.1/amp-delight-player.js

```diff
@@ -0,0 +1,574 @@
+      this.play(false);
+    }
+  }
+
+  /**
+   * @param {!Event} event
+   * @private
+   */
+  handleDelightMessage_(event) {
+    if (event.source !== this.iframe_.contentWindow) {
+      return;
+    }
+
+    const data = objOrParseJson(getData(event));
+    if (data === undefined || data['type'] === undefined) {
+      return; // We only process valid JSON.
+    }
+
+    const {element} = this;
+
+    switch (data['type']) {
+      case DelightEvent.PING: {
+        const guid = data['guid'];
+        if (guid) {
+          this.iframe_.contentWindow./*OK*/postMessage(JSON.stringify(/** @type {JsonObject} */ ({
+            type: DelightEvent.PONG,
+            guid,
+            idx: 0,
+          })), '*');
+        }
+        break;
+      }
+      case DelightEvent.READY: {
+        this.triggerAction_(VideoEvents.LOAD, null);
+        element.dispatchCustomEvent(VideoEvents.LOAD);
+        this.playerReadyResolver_(this.iframe_);
+        break;
+      }
+      case DelightEvent.PLAYING: {
+        this.triggerAction_(VideoEvents.PLAYING, null);
+        element.dispatchCustomEvent(VideoEvents.PLAYING);
+        break;
+      }
+      case DelightEvent.PAUSED: {
+        this.triggerAction_(VideoEvents.PAUSE, null);
+        element.dispatchCustomEvent(VideoEvents.PAUSE);
+        break;
+      }
+      case DelightEvent.ENDED: {
+        this.triggerAction_(VideoEvents.ENDED, null);
+        element.dispatchCustomEvent(VideoEvents.ENDED);
+        break;
+      }
+      case DelightEvent.TIME_UPDATE: {
+        const payload = data['payload'];
+        this.currentTime_ = payload.currentTime;
+        this.playedRanges_ = payload.playedRanges;
+        break;
+      }
+      case DelightEvent.MUTED: {
+        this.triggerAction_(VideoEvents.MUTED, null);
+        element.dispatchCustomEvent(VideoEvents.MUTED);
+        break;
+      }
+      case DelightEvent.UNMUTED: {
+        this.triggerAction_(VideoEvents.UNMUTED, null);
+        element.dispatchCustomEvent(VideoEvents.UNMUTED);
+        break;
+      }
+      case DelightEvent.DURATION: {
+        const payload = data['payload'];
+        this.totalDuration_ = payload.duration;
+        break;
+      }
+      case DelightEvent.EXPANDED: {
+        this.setFullHeight_();
+        break;
+      }
+      case DelightEvent.MINIMIZED: {
+        this.setInlineHeight_();
+        break;
+      }
+      case DelightEvent.ENTERED_FULLSCREEN: {
+        this.isFullscreen_ = true;
+        break;
+      }
+      case DelightEvent.EXITED_FULLSCREEN: {
+        this.isFullscreen_ = false;
+        break;
+      }
+    }
+  }
+
+  /**
+   * Triggers a video event.
+   * @param {string} name
+   * @param {?Object} detail
+   * @private
+   */
+  triggerAction_(name, detail) {
+    const {element} = this;
+    const event =
+        createCustomEvent(this.win, `${TAG}.${name}`,
+            dict({'response': detail}));
+    const actions = Services.actionServiceForDoc(this.element);
```

Comment:
Please just use `element.dispatchCustomEvent`. Using `actionServiceForDoc` would expose all of these to the outside world. We already selectively expose them via video-manager, no need for you to do that. We only expose a few and as `LOW-TRUST` events, we can not expose them as HIGH_TRUST since they are not directly tied to a user action.